### PR TITLE
Tikz support

### PR DESCRIPTION
**Pull request by PatrickMassot**
This PR brings support for tikz and tikz-cd packages when using the HTML5 renderer. This was already discussed quite a bit in issue #45 and PR #62. See also the documentation I wrote.

I guess there should be a way to do that using the Imager mechanism but I had no success at all along this path. There were too many things I didn't understand. The current implementation works for me but feel free to redo it the right way.

Last note: this package and its tests are fully compatible with python3. If this is merged then I'll be able to fill a python3 PR instantly, using only cherry-picked commits.

### CODE REVIEW

File: plasTeX/Packages/tikz.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        object_id = env.attrs['id']
+        basepath = os.path.join(tmp_dir, object_id)
+        texpath = basepath + '.tex'
+        pdfpath = basepath + '.pdf'
+        svgpath =  basepath + '.svg'
+
+        context = {'tikzpicture': env.text.strip() }
+        template.stream(**context).dump(texpath, encoding)
+
+        os.system(compiler + ' ' + texpath)
```
Comment (kesmit13) [inline plasTeX/Packages/tikz.py]: Can these os.system calls be replaced with subprocess.call?  If the paths contain spaces, this could fail.

Comment (PatrickMassot) [inline plasTeX/Packages/tikz.py]: Sure. I'll do that in the evening.

**Comment by PatrickMassot** [comment]
I fixed this issue. I also took the opportunity to add a true functional test actually calling the main plastex script, pdflatex and pdf2svg. It turned out to be a good idea. In my real world examples the output directory was always set and I didn't realize that the default value needed template interpolation. And also the images directory may not exist in a super simple example...

**Comment by PatrickMassot** [comment]
cherry-picked

**Pull request closed without merging**